src/mlib/int128.test.cpp

```diff
@@ -0,0 +1,305 @@
+#include "./int128.h"
+
+#include <iostream>
+#include <random>
+#include <string>
+
+#if (defined(__GNUC__) && __GNUC__ < 7) || \
+   (defined(_MSC_VER) && _MSC_VER < 1920)
```

Comment:
```suggestion
#if (!defined(__clang__) && defined(__GNUC__) && __GNUC__ < 7) || \
   (defined(_MSC_VER) && _MSC_VER < 1920)
```
This might be unnecessarily ruling out Clang according to local tests where my Clang 10 compiler is reporting `__GNUC__ == 4`.

Reply:
[pain.](https://stackoverflow.com/questions/38499462/how-to-tell-clang-to-stop-pretending-to-be-other-compilers)

Let's just hope that GCC never decides to "pretend" to be Clang as well.